homeassistant/components/mobile_app/webhook.py

```diff
@@ -12,71 +11,103 @@
+from homeassistant.helpers import device_registry as dr
 from homeassistant.helpers.discovery import load_platform
-from homeassistant.helpers.storage import Store
+from homeassistant.helpers.template import attach
 from homeassistant.helpers.typing import HomeAssistantType
 
 from .const import (ATTR_ALTITUDE, ATTR_APP_COMPONENT, ATTR_BATTERY,
                     ATTR_COURSE, ATTR_DEVICE_ID, ATTR_DEVICE_NAME,
                     ATTR_EVENT_DATA, ATTR_EVENT_TYPE, ATTR_GPS,
-                    ATTR_GPS_ACCURACY, ATTR_LOCATION_NAME, ATTR_SPEED,
+                    ATTR_GPS_ACCURACY, ATTR_LOCATION_NAME, ATTR_MANUFACTURER,
+                    ATTR_MODEL, ATTR_OS_VERSION, ATTR_SPEED,
                     ATTR_SUPPORTS_ENCRYPTION, ATTR_TEMPLATE,
                     ATTR_TEMPLATE_VARIABLES, ATTR_VERTICAL_ACCURACY,
                     ATTR_WEBHOOK_DATA, ATTR_WEBHOOK_ENCRYPTED,
                     ATTR_WEBHOOK_ENCRYPTED_DATA, ATTR_WEBHOOK_TYPE,
-                    CONF_SECRET, DATA_DELETED_IDS, DATA_REGISTRATIONS, DOMAIN,
+                    CONF_SECRET, DATA_DELETED_IDS, DOMAIN,
                     ERR_ENCRYPTION_REQUIRED, WEBHOOK_PAYLOAD_SCHEMA,
                     WEBHOOK_SCHEMAS, WEBHOOK_TYPE_CALL_SERVICE,
                     WEBHOOK_TYPE_FIRE_EVENT, WEBHOOK_TYPE_RENDER_TEMPLATE,
                     WEBHOOK_TYPE_UPDATE_LOCATION,
                     WEBHOOK_TYPE_UPDATE_REGISTRATION)
 
 from .helpers import (_decrypt_payload, empty_okay_response, error_response,
-                      registration_context, safe_registration, savable_state,
-                      webhook_response)
+                      get_config_entry, registration_context,
+                      safe_registration, webhook_response)
 
 
 _LOGGER = logging.getLogger(__name__)
 
 
-def register_deleted_webhooks(hass: HomeAssistantType, store: Store):
+def register_deleted_webhooks(hass: HomeAssistantType):
     """Register previously deleted webhook IDs so we can return 410."""
     for deleted_id in hass.data[DOMAIN][DATA_DELETED_IDS]:
         try:
             webhook_register(hass, DOMAIN, "Deleted Webhook", deleted_id,
-                             partial(handle_webhook, store))
+                             handle_webhook)
         except ValueError:
             pass
 
 
-def setup_registration(hass: HomeAssistantType, store: Store,
-                       registration: Dict) -> None:
+async def setup_registration(hass: HomeAssistantType,
+                             registration: Dict,
+                             entry: ConfigEntry = None) -> bool:
     """Register the webhook for a registration and loads the app component."""
+    device_registry = await dr.async_get_registry(hass)
+
+    identifiers = {
+        (ATTR_DEVICE_ID, registration[ATTR_DEVICE_ID]),
+        (CONF_WEBHOOK_ID, registration[CONF_WEBHOOK_ID])
+    }
+
+    if entry is None:
+        entry = await get_config_entry(hass, registration[CONF_WEBHOOK_ID])
```

Comment:
dude! You are passing `entry.data` to this function in `async_setup_entry`, just pass `entry` 🤣